src/Definition/Teradata.php

```diff
@@ -677,6 +677,30 @@ public function getBasetype(): string
 
     public static function getTypeByBasetype(string $basetype): string
     {
-        throw new LogicException('Method is not implemented yet.');
+        $basetype = strtoupper($basetype);
+
+        if (!BaseType::isValid($basetype)) {
+            throw new InvalidTypeException(sprintf('Base type "%s" is not valid.', $basetype));
+        }
+
+        switch ($basetype) {
+            case BaseType::BOOLEAN:
+                return self::TYPE_BYTEINT;
```

Comment:
Teradata [nema](https://support.teradata.com/knowledge?id=kb_article_view&sys_kb_id=7abc88404738191486f3405c346d4377) `boolean` type, ale nahrazuje se to za `byteint`, pripadne se vytvari [custom type](https://forgetcode.com/teradata/1865-boolean-type).
Nechal bych tam tenhle fallback.

Pripadne muzeme vyhodit chybu, ze TD nepodporuje Boolean, ale to mi prijde hloupy...
```php
            case BaseType::BOOLEAN:
                throw new InvalidTypeException('Base type "%s" is not supported.', $basetype);
```